File: SelectionSort/SelectionSort.php
<?php

declare(strict_types=1);

$arr = [1,5,3,6,2,10];

function findSmallest(array $arr): int
{
    $smallest = $arr[0];
    $smallestIndex = 0;
    foreach ($arr as $key => $item) {
        if ($item < $smallest) {
            $smallest = $item;
            $smallestIndex = $key;
        }
    }

    return $smallestIndex;
}

function selectionSort(array $arr): array
{
    $result = [];
    foreach ($arr as $ignored) {
        $smallest = findSmallest($arr);
        $result[] = $arr[$smallest];
        unset($arr[$smallest]);
        $arr = array_values($arr);
    }

    return $result;
}

var_dump(selectionSort($arr));

File: SelectionSort/SelectionSort.go
package main

import "fmt"

func main() {
	arr := []int{1, 5, 3, 6, 2, 10}
	fmt.Println(selectionSort(arr))
}

func findSmallest(arr []int) int {
	smallest := arr[0]
	smallestIndex := 0
	for index, item := range arr {
		if item < smallest {
			smallest = item
			smallestIndex = index
		}
	}

	return smallestIndex
}

func selectionSort(arr []int) []int {
	result := []int{}
	for _, item := range arr {
		fmt.Println(item)
		smallest := findSmallest(arr)
		result = append(result, arr[smallest])
		arr = removeIndex(arr, smallest)
	}

	return result
}

func removeIndex(s []int, index int) []int {
	return append(s[:index], s[index+1:]...)
}
